content/blog/wireshark-protobuf-and-grpc-dissectors.md

```diff
@@ -0,0 +1,243 @@
+
+- Wireshark 2.6.0 - adds the initial version of Protobuf and gRPC dissectors.
+
+- Wireshark 3.2.0 - supports dissecting serialized Protobuf data (such as gRPC) more precisely according to *.proto files, and supports dissecting gRPC messages of streaming calls.
+
+- Wireshark 3.3.0 - fixes some bugs about parsing *.proto files and supports some new features about Protobuf dissector.
+
+## Features
+
+The main features about Wireshark Protobuf and gRPC dissectors includes:
+
+- Supports dissecting gRPC messages serialized in Protobuf or JSON format.
+
+- Supports dissecting the gRPC messages of unary, server streaming, client streaming, and bidirectional streaming RPC calls.
+
+- Message and service definition language files (*.proto) can be configured to enable more precise parsing of the serialized data in Protobuf format for protocols such as gRPC.
+
+- Protobuf fields can be dissected as Wireshark fields that allows user to input the full name of a Protobuf field or message in the filter toolbar to search packages containing the field or message.
+
+- You can register your own subdissectors in the 'protobuf_field' dissector table, which is keyed with the full names of the fields, for further parsing Protobuf fields of BYTES or STRING type.
+
+- You can add the mapping record between UDP port and Protobuf message type in the Protobuf protocol preferences dialog, so that the Wireshark can decode the traffic on that UDP port as this Protobuf message.
+
+- You can create your own dissectors to parsing the protocols based on Protobuf over UDP or TCP.
+
+## Examples about using Protobuf and gRPC dissectors
+
+There are some examples to show how to use the Wireshark Protobuf and gRPC dissectors. You can get more details about these examples from [Protobuf wiki](https://gitlab.com/wireshark/wireshark/-/wikis/Protobuf) and [gRPC wiki](https://gitlab.com/wireshark/wireshark/-/wikis/gRPC) pages from Wireshark official website.
+
+### Sample .proto files
+
+These .proto files will be used in the following examples.
+
+The content of `addressbook.proto`:
+
+```protobuf
+// This file comes from the official Protobuf example with a little modification.
+syntax = "proto3";
+package tutorial;
+import "google/protobuf/timestamp.proto";
+
+message Person {
+  string name = 1;
+  int32 id = 2;  // Unique ID number for this person.
+  string email = 3;
+
+  enum PhoneType {
+    MOBILE = 0;
+    HOME = 1;
+    WORK = 2;
+  }
+
+  message PhoneNumber {
+    string number = 1;
+    PhoneType type = 2;
+  }
+
+  repeated PhoneNumber phone = 4;
+  google.protobuf.Timestamp last_updated = 5;
+  bytes portrait_image = 6;
+}
+
+message AddressBook {
+  repeated Person people = 1;
+}
+```
+
+According to the line `'import "google/protobuf/timestamp.proto"'`, the file `addressbook.proto` depends on the official *.proto library of Protobuf.
```

Comment:
the "official *.proto library of Protobuf"  is technically called "well known types", timestamp WKT is one of them.

https://developers.google.com/protocol-buffers/docs/reference/google.protobuf

Reply:
@huangqiangxiong, will you be making this change?

Reply:
@chalin, I was going to make this change after collecting comments from all three of you. Do you mean I can change it at any time?